quickwit-config/src/config.rs

```diff
@@ -280,69 +280,77 @@ impl QuickwitConfig {
     }
 
-    /// Returns the gossip listen port
-    /// extracted from config if specified or the same as `rest_listen_port`.
-    fn gossip_listen_port(&self) -> u16 {
-        match self.gossip_listen_port {
-            Some(grpc_listen_port) => grpc_listen_port,
-            // By default, we use the same port number as the rest port but this is UDP.
-            None => self.rest_listen_port,
+    /// Returns the gRPC listen address of the node, i.e. the socket address on which the gRPC
+    /// service listens for TCP connections.
+    pub async fn grpc_listen_addr(&self) -> anyhow::Result<SocketAddr> {
+        get_socket_addr(&(self.listen_address.as_str(), self.grpc_listen_port())).await
+    }
+
+    /// Returns the gRPC public address of the node, i.e. the socket address to connect to in order
+    /// to send gRPC requests to the node.
+    pub async fn grpc_public_addr(&self) -> anyhow::Result<SocketAddr> {
+        match env::var("QW_PUBLIC_IP") {
+            Ok(ip_addr) => get_socket_addr(&(ip_addr, self.grpc_listen_port())).await,
+            Err(_) => self.grpc_listen_addr().await,
         }
     }
 
-    /// Returns the grpc socket address.
-    pub fn grpc_socket_addr(&self) -> anyhow::Result<SocketAddr> {
-        get_socket_addr(&(self.listen_address.as_str(), self.grpc_listen_port()))
+    /// Returns the gossip listen port of the node (UDP).
+    /// Extracted from the config if specified or same as `rest_listen_port` otherwise.
+    fn gossip_listen_port(&self) -> u16 {
+        // By default, we use the same port number as the REST port but UDP this time.
+        self.gossip_listen_port.unwrap_or(self.rest_listen_port)
     }
 
-    /// Returns the gossip socket address.
-    pub fn gossip_socket_addr(&self) -> anyhow::Result<SocketAddr> {
-        get_socket_addr(&(self.listen_address.as_str(), self.gossip_listen_port()))
+    /// Returns the gossip listen address of the node, i.e. the UDP socket address on which the node
+    /// receives gossip messages.
+    pub async fn gossip_listen_addr(&self) -> anyhow::Result<SocketAddr> {
+        get_socket_addr(&(self.listen_address.as_str(), self.gossip_listen_port())).await
     }
 
-    /// The node gossip_public_address should ideally be specified via config;
-    /// environment variable interpolation if necessary.
-    /// right now we just try to fetch from environment variable otherwise fallback
-    /// to listen_address.
-    pub fn gossip_public_addr(&self) -> anyhow::Result<SocketAddr> {
-        match env::var("QW_GOSSIP_PUBLIC_ADDRESS") {
-            Ok(addr) => Ok(addr.parse::<SocketAddr>()?),
-            Err(_) => self.gossip_socket_addr(),
+    /// Returns the gossip public address of the node, i.e. the socket address to send UDP packets
+    /// to in order to gossip with the node.
+    pub async fn gossip_public_addr(&self) -> anyhow::Result<SocketAddr> {
+        match env::var("QW_PUBLIC_IP") {
```

Comment:
That should be in Quickwit config I think. Can we open a ticket to move that in the quickwti config? (with the same CLI override temporary fix as you introduced in this PR for other options)

Reply:
Yes, I agree, this is a quick fix.